### ISSUE TITLE
Remove support for X-Reported-With header

### ISSUE THREAD

**Issue by miketaylr** [webcompat/webcompat.com#1254]
And only support a `src=` param, to make things simple. Before we can do this we need to update all the add-ons and get them published.

**Comment by karlcow** [comment]
How do we treat legacy? Aka add-on deployed in the wild but not updated.

**Comment by karlcow** [comment]
Also I guess this is linked to #1249 or am I confused 
`X-Requested-With` here in #1254 and `X-Reported-With` in #1249

**Comment by miketaylr** [comment]
> How do we treat legacy? Aka add-on deployed in the wild but not updated.

So, it turns out I never published the addons that sent `X-Reported-With` -- I dropped the ball on that. I did just publish versions 0.5 of Firefox Desktop and Chrome that send `&src=reporter-addon-chrome`, etc. yesterday.

The odds that someone built the addon from the repo for the few weeks that code existed are probably low enough to not worry about.

(Also, I just keep mixing it up, it's actually just `X-Reported-With`)
